internal/controllers/securitygroup/actuator.go

```diff
@@ -241,6 +219,115 @@ func (actuator securityGroupActuator) updateTags(ctx context.Context, orcObject
 	return nil, orcObject, osResource, err
 }
 
+func rulesMatch(orcRule *orcv1alpha1.SecurityGroupRule, osRule *rules.SecGroupRule) bool {
+	// Don't compare description if it's not set in the spec
+	if orcRule.Description != nil && string(*orcRule.Description) != osRule.Description {
+		return false
+	}
+
+	// Don't compare direction if it's not set in the spec.
+	// TODO check what we get from neutron in this field if we didn't set it in the spec
+	if orcRule.Direction != nil && string(*orcRule.Direction) != osRule.Direction {
+		return false
+	}
+
+	// Always compare protocol. Unset == "" from gophercloud
+	if string(ptr.Deref(orcRule.Protocol, "")) != osRule.Protocol {
+		return false
+	}
+
+	if string(orcRule.Ethertype) != osRule.EtherType {
+		return false
+	}
+
+	if orcRule.PortRange == nil {
+		if osRule.PortRangeMin != 0 || osRule.PortRangeMax != 0 {
```

Comment:
I think we need to be a bit smarter here, because the openstack resource might have the port set to 0 and still meaning it needs to do filtering when the protocol is icmp.

We need an extra check that `osRule.Protocol != "icmp"` I think.

Reply:
I don't see it. These values need to match regardless of protocol. If the PortRange struct is nil in ORC, that translates to min==0, max==0 in gophercloud. This is just an equality check.

Reply:
Yep, I think you're right.